src/main/features/response/form/models/statement-of-means/dependants.ts

```diff
@@ -48,13 +32,11 @@ export class Dependants implements Serializable<Dependants> {
 
     const dependants = new Dependants(
       value.hasAnyChildren !== undefined ? toBoolean(value.hasAnyChildren) === true : undefined,
-      toNumberOrUndefined(value.under11),
-      toNumberOrUndefined(value.between11and15),
-      toNumberOrUndefined(value.between16and19)
+      NumberOfChildren.fromObject(value.numberOfChildren)
     )
 
     if (!dependants.hasAnyChildren) {
-      dependants.under11 = dependants.between11and15 = dependants.between16and19 = undefined
+      dependants.numberOfChildren = undefined
```

Comment:
You can reverse the condition and deserialise `NumberOfChildren` only when `dependants.hasAnyChildren` evaluates to `true`. This way deserialisation method is not called when unneeded.

Reply:
ok

Reply:
refactored